src/coreclr/src/vm/comdelegate.cpp

```diff
@@ -1146,6 +1146,9 @@ PCODE COMDelegate::ConvertToCallback(MethodDesc* pMD)
     }
     CONTRACTL_END;
 
+    // Check the method for proper usage.
+    ThrowIfInvalidUnmanagedCallersOnlyUsage(pMD);
+
```

Comment:
Fetching of the calling convention below can be probably moved to `UMThunkMarshInfo::RunTimeInit` as well, and combined with the other check.

Reply:
I don't see how this calling convention is even being used actually. It seems to be immediately overwritten in https://github.com/dotnet/runtime/blob/bddbb03b33162a758e99c14ae821665a647b77c7/src/coreclr/src/vm/dllimportcallback.cpp#L724

The `UMThunkMarshInfo::CompileNExportThunk()` is always called in `UMThunkMarshInfo::RunTimeInit()` and no one seems to call `UMThunkMarshInfo::GetCallingConvention()`.

Reply:
Yep. I just removed `UMThunkMarshInfo::GetCallingConvention()` from the `x86` and `x64` build without issue. It looks like setting that value is over-written. That is something I will need to take a look at.

Reply:
I am more convinced the `x86` scenario is busted. I don't see how the user specified calling convention in the `UnmanagedCallersOnly` attribute is respected. The only value that seems to be consumed is from the sig, not the attribute value.